compiler/noirc_evaluator/src/ssa/acir_gen/mod.rs

```diff
@@ -1258,16 +1258,15 @@ impl Context {
         // The return value may or may not be an array reference. Calling `flatten_value_list`
         // will expand the array if there is one.
         let return_acir_vars = self.flatten_value_list(return_values, dfg);
+        let mut warnings = Vec::new();
         for acir_var in return_acir_vars {
-            // TODO(Guillaume) -- disabled as it has shown to break
-            // TODO with important programs. We will add it back once
-            // TODO we change it to a warning.
-            // if self.acir_context.is_constant(&acir_var) {
-            //     return Err(InternalError::ReturnConstant { call_stack: call_stack.clone() });
-            // }
+            if self.acir_context.is_constant(&acir_var) {
+                warnings
+                    .push(InternalError::ReturnConstant { call_stack: call_stack.clone() }.into());
```

Comment:
I think `ReturnConstant` being an `InternalError` is a mistake. `InternalError` basically represents an ICE but a user compiling a program which has a constant return value is definitely not an ICE.

We should add a separate enum for warnings in my opinion.

Reply:
That said, this preexists this PR so it's not a blocker.

Reply:
I changed to use a Warning instead, and a SsaReport enum that I will use to add info diagnostics for another issue.